### ISSUE TITLE
Sometimes emails arrive to wrong email boxes

### ISSUE THREAD

**Issue by vladikoff** [mozilla/restmail.net#21]
cc @jrgm @rfk 

See http://restmail.net/mail/pushtest

![image](https://cloud.githubusercontent.com/assets/128755/26833563/d1f74310-4aa0-11e7-8623-07a0f96606ce.png)

`signin0.7721520143095404%40restmail.net` arrived to the `pushtest` mailbox

@jrgm has more information on this. Seems like a bug in restmail updates

**Comment by jrgm** [comment]
So, one thing that is happening is (from tracing raw network traffic w/ ngrep) is that more than one email message can be delivered on a connection. So for a sender sequence like:
```
Connect
EHLO
MAIL FROM: msgid1
RCPT TO: mail1@
DATA
blah, blah, blah
NOOP. (gets 502 Not implemented)
MAIL FROM: msgid2
RCPT TO: mail2@
DATA
blah, blah, blah
QUIT (gets 221 Bye!)
```
we'll log:
```
[2017-06-06T14:56:37.688Z] Handling SMTP request
[2017-06-06T14:56:38.084Z] Received message for [ 'sync0.6161709907464683@restmail.net' ]
[2017-06-06T14:56:55.733Z] Received message for [ 'sync0.6161709907464683@restmail.net',
  'sync0.5953627610579133@restmail.net' ]
```

Need to poke at this more, but thought I'd pass this along. And I don't know if this is the core problem reported in this issue.

**Comment by jrgm** [comment]
Check out http://restmail.net/mail/sync0.6161709907464683@restmail.net and
http://restmail.net/mail/sync0.5953627610579133@restmail.net. (Note: there may have been other mails to these addresses not shown above, but http://restmail.net/mail/sync0.6161709907464683@restmail.net's mailbox has mail for sync0.5953627610579133@restmail.net (no CC: involved)). Maybe need to handle the envelope RCPT TO, to handle these multiple payloads per connection cleanly.

**Comment by jrgm** [comment]
p.s., I can switch restmail back to the previous release, if needed.

**Comment by rfk** [comment]
> So, one thing that is happening is (from tracing raw network traffic w/ ngrep)
> is that more than one email message can be delivered on a connection.

Just to confirm, I tried this out by hand and it did indeed cause restmail to deliver both emails to both addresses, rather just one to each address.  So it seems plausible that this is the cause of the bug.  My proposed fix is in #22 and was actually a pretty small change overall.
